### PR TITLE
Fix ONNX graph missing input and with extra output constants

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
Currently, when models are exported to ONNX although they are semantically correct, all model inputs are exported as Constants, which results in user input to be ignored during the model execution by a Runtime.

The root cause is actually the way the model was created as opposed opposed to a bug on the ONNX converter. This PR rewrites parts of the model to replace some "python" APIs by their "PyTorch" counterparts, that can handle Tensors tracing properly.

### CODE REVIEW

**Comment by thiagocrepaldi** [comment]
@garymm I left as draft until #4120  #4205 are merged, but this is a merge candidate in case you want to review it

**Review by FrancescoMandru — APPROVED** [review]

File: detectron2/export/flatten.py [hunk 1]
```diff
@@ -307,6 +322,18 @@ def forward(self, *args: torch.Tensor):
...
+                if len(flattened_output_tensors) > len(flattened_outputs) and\
+                        self.__is_in_torch_onnx_export:
+                    warnings.warn("PyTorch ONNX export (`torch.onnx.export`) detected!"
+                                  " To prevent extra outputs in the ONNX graph, the original"
+                                  " model output cannot be reconstructed through"
+                                  " `adapter.outputs_schema(flattened_outputs)`."
+                                  " For results evaluation, use `torch.jit.trace` instead.")
+                    flattened_outputs = flattened_outputs[:len(outputs)]
+
                 if self.outputs_schema is None:
```
Comment (FrancescoMandru) [inline detectron2/export/flatten.py:337]: Does it means that the final output will need to be processed to be well formatted and ready to use?

File: tests/test_export_onnx.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    @unittest.skipIf(not min_torch_version("1.11.1"), "PyTorch 1.11.1+ is the minimum version")
+    def testMaskRCNNFPN_with_postproc(self):
+        def inference_func(model, image):
+            inputs = [{"image": image, "height": image.shape[1], "width": image.shape[2]}]
+            return model.inference(inputs, do_postprocess=True)[0]["instances"]
+
+        self._test_model_zoo_from_config_path(
+            "COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml", inference_func, opset_version=16
+        )
+
```
Comment (thiagocrepaldi) [inline tests/test_export_onnx.py:92]: @FrancescoMandru does that answer the question about the post processing?

Comment (FrancescoMandru) [reply tests/test_export_onnx.py:92]: Yes @thiagocrepaldi, thank you!

**Pull request closed without merging**